### ISSUE TITLE
How to dev two packages at once (package + consumer)?

### ISSUE THREAD

**Issue by Apidcloud** [kentcdodds/ama#319]
Hello again!

I thought it might be better to create another issue for this question.

It's somewhat related with the previous one though (#318)

Say you are developing a package that's not yet finished (alpha or beta) - let's call it **BETA**. Also say you are developing a client app that consumes **BETA** -- let's call it **CONSUMER**.

What would you recommend to get the workflow right? I mean, should I keep copying the **BETA** build to **CONSUMER** every time I want to test something within it? Or should I publish **BETA** to npm under a tag (#318) and use that on **CONSUMER**?

I really don't know if there's a proper way to accomplish this and thus creating this issue.

Thanks for your time once again!

**Comment by AugustinLF** [comment]
Hmmm, usually [symlinks](https://docs.npmjs.com/cli/link), using an `npm link` (or `yarn link)` will work for you :) This lets you make consumer believe that there is a `beta` package in its `node_modules`.

**Comment by Apidcloud** [comment]
Thanks, @AugustinLF ! Seems like a reasonable solution. Everyone would need to `npm link` locally when developing right?

**Comment by AugustinLF** [comment]
Yep, you're right. The good thing, is that you have to do it only once :)

And yes, you can consider the link as if you were directly developing in `consumer/node_modules/beta` :) The only caveat (but that's shared with independant packages) is that your `watch`es on the `consumer` side might not work properly. For instance, on my app, webpack does update properly when `beta` is updated, but my flow server picks up  the old definitions of `beta` whenever I change them (I have to reload the server).

**Comment by Apidcloud** [comment]
Marvelous 👓 Rebuilding **`beta`** and refreshing the **`consumer`** seems to do the trick!

When releasing a new version of the **`consumer`** I just need to release **`beta`** and update the package back on **`consumer's`** package.json then?

I'll keep the issue open in case @kentcdodds wants to add something.

Thanks a lot!

**Comment by kentcdodds** [comment]
Hi @Apidcloud!

Thanks so much for stepping in here @AugustinLF! That's what I would suggest as well. I've never had a case where other folks needed to `npm link` something that I was working on. I pretty much work with `npm link` and when it's ready I'll publish a `beta` (or just release it).

Good luck!

**Comment by Apidcloud** [comment]
Thank you! 👍

**Comment by Apidcloud** [comment]
Hey! I just came across something that can also be used: [Git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules)
I'm still exploring it, but seems really handy! 👍

**Comment by kentcdodds** [comment]
I've not experienced it myself, but I've heard from many others that submodules promise great things but are a huge pain for some reason. Proceed with caution.
